### ISSUE TITLE
Add a --stdin-filename option

### ISSUE THREAD

**Issue by w0rp** [troessner/reek#1296]
[ALE](https://github.com/w0rp/ale) wants to check for problems with Vim buffers while you type, and doing so requires checking text via stdin, or temporary files. This poses a problem when you want to write a configuration file for ignoring some rules for certain files, etc. Other tools like ESLint solve this problem by allowing you to specify the path to a file being checked when the text is being checked via stdin, so you can apply whatever is in configuration files and also check for problems while you type.

I propose adding an `--stdin-filename` option so you can write `reek --stdin-filename=foo/bar.rb < tempoary_file` or similar, so you can both check for problems while you type and apply settings from configuration files specific to the files.

**Comment by troessner** [comment]
Interesting problem. Makes sense to me. I'll put this on my todo list but can't promise you when I'll get around to implementing this, unless @mvz or @chastell want to take over.

**Comment by mvz** [comment]
I do wonder if there are other solutions to this problem, since I find it rather ugly.

**Comment by mvz** [comment]
I'm also wondering if this shouldn't be generalized to (option name to be discussed):

```bash
reek --pretend-filename foo/bar.rb temporary_file.rb
```

**Comment by mvz** [comment]
As for one other solution to this problem, no less ugly:

```bash
reek --extract-config-for foo/bar.rb > /tmp/temporary_config.reek
reek --config /tmp/temporary_config.reek < /tmp/temporay_file.rb
```

**Comment by w0rp** [comment]
Before you go looking for other options, you should look at how this problem has been handled in the past. The `--stdin-filename` option for ESLint and some similar tools solves the problem pretty well. Bike-shedding over the option name is what other projects spend half their time on.

**Comment by troessner** [comment]
It's quite a bit inappropriate and insulting for you to insinuate that we are bike shedding here. You havent contributed anything to Reek and if we want to discuss this option name then this is none of your business. You might want to reflect on your attitude.

**Comment by w0rp** [comment]
Take it easy. I said other projects spend most of their time on bike-shedding about an argument name, and that's just how it goes. I'm amazed you get that worked up for such an innocuous comment.

**Comment by troessner** [comment]
It wasnt innocuous and your defensive arrogance is starting to get annoying. This conversation is over. Time for you to get lost now.

**Comment by paul** [comment]
Jeez, calm down a bit. @w0rp's comment seemed perfectly reasonable, and emulating other projects that came across similar problems is worth a ton. Perhaps @w0rp hasn't contributed to Reek directly, but 99% of my usage of it comes from the tool that they maintain (w0rp/ALE). I think its worth considering their words from that point of view, rather than assuming the worst that they're trying to be insulting.

I came into this thread because I had some time to perhaps work on this issue and become a new Reek contributor, but now I'm concerned about how it would be received.

**Comment by w0rp** [comment]
I just want everyone to get along. I'm sorry if I sounded rude before.

**Comment by troessner** [comment]
@paul for sure every contribution will be more than welcome - nothing to worry about.
@w0rp I'm sorry as well, let's start over.

**Comment by w0rp** [comment]
Okay, sounds good to me. :+1:

**Comment by troessner** [comment]
@paul are you still up for giving this a shot? If not we'd probably look into this after the release of [Reek 5](https://github.com/troessner/reek/issues/1305).

**Comment by troessner** [comment]
Nice :)

**Comment by troessner** [comment]
@w0rp we just released Reek 5 via #1324 with the --stdin feature included, you should be able to pick it up from here ;)

**Comment by w0rp** [comment]
Okay, sounds good to me. :+1:
